dialog_lib/agents/abstract.py

```diff
@@ -103,6 +104,74 @@ def messages(self):
         return self.memory.messages
 
 
+class AbstractLcelClass(AbstractLLM):
+
+
+    @property
+    def chat_model(self):
+        """
+        builds and returns the chat model for the LCEL
+        """
+        raise NotImplementedError("Chat model must be implemented")
+
+    @property
+    def context_dict(self):
+        """
+        builds and returns the context dictionary for the LCEL
+        """
+        raise NotImplementedError("Context dictionary must be implemented")
+
+    @property
+    def chain(self):
+        """
+        builds and returns the chain for the LCEL
+        """
+        return (
+            self.context_dict | self.prompt | self.chat_model
+        )
+
+    @property
+    def get_memory_instance(self):
```

Comment:
since this is a method that returns the memory instance and has a verb in its name, maybe it could be called as plain`memory` (a more property-like name), what do you think?